File: atcoder/abc236A_chukodai.nim
# uninhm
# https://atcoder.jp/contests/abc236/tasks/abc236_a
# implementation

import strutils, sequtils

var s = stdin.readLine.strip
var a, b: int
(a, b) = stdin.readLine.split.map parseInt

swap s[a-1], s[b-1]

echo s

File: atcoder/abc236a_chukodai.hs
-- vicfred
-- https://atcoder.jp/contests/abc236/tasks/abc236_a
-- implementation
main :: IO ()
main = do
  s <- getLine
  [a, b] <- map read . words <$> getLine :: IO [Int]
  putStrLn $ take (a - 1) s ++ [s !! (b - 1)] ++ drop a (take (b - 1) s) ++ [s !! (a - 1)] ++ drop b s
